### PR TITLE
Bugfix FXIOS-5965 [v112] Ensure main thread on cancelling passwords filter

### PR DESCRIPTION
**Pull request by nishant2718**
# [FXIOS-5965](https://mozilla-hub.atlassian.net/browse/FXIOS-5965) | #13558

### Description
Prevent app crash by ensuring main. There was previously mismatching thread priority of some work that was causing a crash.

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [x] Documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by nishant2718** [comment]
Seems like an unrelated test fails, `OnboardingCardViewModelTests.testSendOnboardingPrimaryTap_SyncCard()`. But it passes locally.

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Project coverage: 30.22%</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Edited 1 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 0 files</td>
    </tr>
  </tbody>
</table>

## Client.app: Coverage: 27.12
| File | Coverage ||
| --- | --- | --- |
LoginListDataSourceHelper.swift | 92.16% | ✅

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against a72d0e297c1a8f4635a09dcd62e60ae81ee84ab0
</p>

File: Client/Frontend/LoginManagement/LoginListDataSourceHelper.swift [hunk 1]
```diff
@@ -62,7 +62,8 @@ class LoginListDataSourceHelper {
         var sections = [Character: [LoginRecord]]()
         var titleSet = Set<Character>()
 
-        DispatchQueue.global(qos: .userInteractive).async { [weak self] in
+        // Due to thread priority mismatches, we need to force this onto main to avoid app crashing. 
+        ensureMainThread { [weak self] in
```
Comment (OrlaM) [inline Client/Frontend/LoginManagement/LoginListDataSourceHelper.swift:66]: I'm not sure I understand why this needs to be the main thread. Does setting the qos to default also crash?

Looking at the code in the function I'm not sure why this is being dispatched to a new thread at all. Can we remove this line and leave it run on whatever thread initiated the request?

Comment (nishant2718) [reply Client/Frontend/LoginManagement/LoginListDataSourceHelper.swift:66]: Yeah let me get back to you on that.

Comment (nishant2718) [reply Client/Frontend/LoginManagement/LoginListDataSourceHelper.swift:66]: Okay so doing either of those leads to a crash. Removing Dispatches onto main in other areas leads to a crash in a `cellForRowAt` inside `LoginDataSource` when a user tries to filter logins / passwords. 

It seems like there's a mix of Dispatches to threads of varying qos in this area that might take more time to track and resolve. But there are comments here and there saying [completion needs to run on main thread](https://github.com/mozilla-mobile/firefox-ios/blob/5a0ed3c1967400758241e1d410853d96199aad6d/Client/Frontend/LoginManagement/LoginListViewModel.swift#L133).

Comment (nishant2718) [reply Client/Frontend/LoginManagement/LoginListDataSourceHelper.swift]: After discussing over chat, we decided to move the ensureMain to a level higher to capture the most cases of timing issues in this context.

**Review by OrlaM — APPROVED** [review]

**Pull request merged**